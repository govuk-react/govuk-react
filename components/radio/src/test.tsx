import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import Radio from '.';

describe('Radio', () => {
  let props;
  const example = 'example';

  const wrapper = <Radio name={example}>{example}</Radio>;

  const wrapperInline = (
    <Radio inline name={example}>
      {example}
    </Radio>
  );

  beforeEach(() => {
    props = {
      name: example,
      children: example,
    };
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('passes `props.children` to the rendered `wrapper` as `children`', () => {
    expect(wrapper.props.name).toBe(props.name);
    expect(wrapper.props.children).toBe(props.children);
  });

  it('matches snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('standard mount');
  });

  it('disabled', () => {
    expect(mount(<Radio disabled>Example</Radio>)).toMatchSnapshot('disabled');
  });

  it('matches snapshot for inline', () => {
    expect(mount(wrapperInline)).toMatchSnapshot('inline mount');
  });

  it('can render with hint text', () => {
    expect(mount(<Radio hint="Hint text">Example with hint text</Radio>)).toMatchSnapshot('hint text');
  });
});
