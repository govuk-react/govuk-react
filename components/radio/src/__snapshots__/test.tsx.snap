// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio can render with hint text: hint text 1`] = `
.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  color: #6f777b;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  position: relative;
  min-height: 40px;
  padding: 0 0 0 40px;
  clear: left;
  margin-bottom: 10px;
}

.c1 {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  cursor: pointer;
  opacity: 0;
  cursor: pointer;
}

.c1:checked + span::after {
  opacity: 1;
}

.c1:focus + span::before {
  outline: 3px solid transparent;
  outline-offset: 3px;
  box-shadow: 0 0 0 4px #ffdd00;
}

.c1 + span {
  opacity: 1;
  pointer-events: auto;
}

.c2 {
  display: inline-block;
  margin-bottom: 0;
  padding: 8px 15px 5px;
  cursor: pointer;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
}

.c2:before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  border: 2px solid black;
  border-radius: 50%;
  background: transparent;
}

.c2:after {
  content: '';
  position: absolute;
  top: 10px;
  left: 10px;
  width: 0;
  height: 0;
  border: 10px solid;
  border-radius: 50%;
  opacity: 0;
}

.c4 {
  display: block;
  padding-left: 15px;
  padding-right: 15px;
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 10px;
  }
}

<Radio
  hint="Hint text"
  inline={false}
>
  <src__Label
    inline={false}
  >
    <label
      className="c0"
    >
      <src__Input
        type="radio"
      >
        <input
          className="c1"
          type="radio"
        />
      </src__Input>
      <src__LabelText>
        <span
          className="c2"
        >
          Example with hint text
        </span>
      </src__LabelText>
      <src__StyledRadioHint>
        <span
          className="c3 c4"
        >
          Hint text
        </span>
      </src__StyledRadioHint>
    </label>
  </src__Label>
</Radio>
`;

exports[`Radio disabled: disabled 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  position: relative;
  min-height: 40px;
  padding: 0 0 0 40px;
  clear: left;
  margin-bottom: 10px;
}

.c1 {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  cursor: pointer;
  opacity: 0;
  cursor: auto;
}

.c1:checked + span::after {
  opacity: 1;
}

.c1:focus + span::before {
  outline: 3px solid transparent;
  outline-offset: 3px;
  box-shadow: 0 0 0 4px #ffdd00;
}

.c1 + span {
  opacity: .5;
  pointer-events: none;
}

.c2 {
  display: inline-block;
  margin-bottom: 0;
  padding: 8px 15px 5px;
  cursor: pointer;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
}

.c2:before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  border: 2px solid black;
  border-radius: 50%;
  background: transparent;
}

.c2:after {
  content: '';
  position: absolute;
  top: 10px;
  left: 10px;
  width: 0;
  height: 0;
  border: 10px solid;
  border-radius: 50%;
  opacity: 0;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 10px;
  }
}

<Radio
  disabled={true}
  inline={false}
>
  <src__Label
    inline={false}
  >
    <label
      className="c0"
    >
      <src__Input
        disabled={true}
        type="radio"
      >
        <input
          className="c1"
          disabled={true}
          type="radio"
        />
      </src__Input>
      <src__LabelText>
        <span
          className="c2"
        >
          Example
        </span>
      </src__LabelText>
    </label>
  </src__Label>
</Radio>
`;

exports[`Radio matches snapshot for inline: inline mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  position: relative;
  min-height: 40px;
  padding: 0 0 0 40px;
  clear: left;
  margin-bottom: 10px;
}

.c1 {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  cursor: pointer;
  opacity: 0;
  cursor: pointer;
}

.c1:checked + span::after {
  opacity: 1;
}

.c1:focus + span::before {
  outline: 3px solid transparent;
  outline-offset: 3px;
  box-shadow: 0 0 0 4px #ffdd00;
}

.c1 + span {
  opacity: 1;
  pointer-events: auto;
}

.c2 {
  display: inline-block;
  margin-bottom: 0;
  padding: 8px 15px 5px;
  cursor: pointer;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
}

.c2:before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  border: 2px solid black;
  border-radius: 50%;
  background: transparent;
}

.c2:after {
  content: '';
  position: absolute;
  top: 10px;
  left: 10px;
  width: 0;
  height: 0;
  border: 10px solid;
  border-radius: 50%;
  opacity: 0;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    float: left;
    clear: none;
    margin-right: 20px;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 10px;
  }
}

<Radio
  inline={true}
  name="example"
>
  <src__Label
    inline={true}
  >
    <label
      className="c0"
    >
      <src__Input
        name="example"
        type="radio"
      >
        <input
          className="c1"
          name="example"
          type="radio"
        />
      </src__Input>
      <src__LabelText>
        <span
          className="c2"
        >
          example
        </span>
      </src__LabelText>
    </label>
  </src__Label>
</Radio>
`;

exports[`Radio matches snapshot: standard mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  position: relative;
  min-height: 40px;
  padding: 0 0 0 40px;
  clear: left;
  margin-bottom: 10px;
}

.c1 {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  cursor: pointer;
  opacity: 0;
  cursor: pointer;
}

.c1:checked + span::after {
  opacity: 1;
}

.c1:focus + span::before {
  outline: 3px solid transparent;
  outline-offset: 3px;
  box-shadow: 0 0 0 4px #ffdd00;
}

.c1 + span {
  opacity: 1;
  pointer-events: auto;
}

.c2 {
  display: inline-block;
  margin-bottom: 0;
  padding: 8px 15px 5px;
  cursor: pointer;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
}

.c2:before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  border: 2px solid black;
  border-radius: 50%;
  background: transparent;
}

.c2:after {
  content: '';
  position: absolute;
  top: 10px;
  left: 10px;
  width: 0;
  height: 0;
  border: 10px solid;
  border-radius: 50%;
  opacity: 0;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 10px;
  }
}

<Radio
  inline={false}
  name="example"
>
  <src__Label
    inline={false}
  >
    <label
      className="c0"
    >
      <src__Input
        name="example"
        type="radio"
      >
        <input
          className="c1"
          name="example"
          type="radio"
        />
      </src__Input>
      <src__LabelText>
        <span
          className="c2"
        >
          example
        </span>
      </src__LabelText>
    </label>
  </src__Label>
</Radio>
`;
